### PR TITLE
Panel Query Options: Support query caching options

### LINKED ISSUE #82311: Panel Query Options: Handle caching options (cache timeout, cache TTL)

### PR DESCRIPTION
**Pull request by dprokop**
Closes https://github.com/grafana/grafana/issues/82311

Approached it using behavior attachable to `VizPanel` to avoid additional wrappers around the panels. Think this turned out pretty simple.

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts [hunk 1]
```diff
@@ -295,6 +296,15 @@ function vizPanelDataToPanel(
     ];
   }
 
+  if (cachingOptions?.state.cacheTimeout) {
+    // @ts-expect-error Graphite-specific option, not sure we should add it to schema
+    panel.cacheTimeout = cachingOptions.state.cacheTimeout;
+  }
+  if (cachingOptions?.state.queryCachingTTL) {
+    // @ts-expect-error enterprise/cloud option, not sure we should add it to schema
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts:304]: Thoughts of putting this into the schema...?

File: public/app/features/dashboard-scene/scene/DashboardScene.test.tsx [hunk 2]
```diff
@@ -315,6 +328,20 @@ function buildTestScene(overrides?: Partial<DashboardSceneState>) {
...
         }),
+        new SceneGridItem({
+          body: new VizPanel({
+            title: 'Panel with caching options',
+            key: 'panel-3',
+            pluginId: 'table',
+            $behaviors: [
+              new PanelQueryCachingOptionsBehavior({
+                cacheTimeout: '60',
+                queryCachingTTL: 200000,
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: @dprokop Is there a good reason why these options are not part of SceneQueryRunner?

Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: @torkelo  not really. But I'm sceptical on putting non-oss (`queryCachingTTL`) options in the library... Same with `cacheTimeout` which is used by Graphite data source only.

Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: @dprokop but they are part of DataQueryRequest so I think it's fine, and I think scene apps would like to be able to use query caching as well

Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: Agree with `queryCachingTTL` which is a cross-data source option and makes sense in the apps. But I think I would stick with the core-only behavior for the Graphite's `cacheTimeout`. Feels non-desirable to me to put data source-specific API into the library.

Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: Or maybe I'm wrong and there are more data sources that use `cacheTimeout`...? I guess some OSS ones may use it.

Comment (torkelo) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: @dprokop it's part of plugin.json data source feature definition, I know zabbix uses it. A bit sad that the enterprise cachingTTL did not use that feature and built a separate option

Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.test.tsx:339]: kk, then let's move it to SQR :)

**Comment by dprokop** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/82448

**Pull request closed without merging**